assembly/common/src/main/resources/bin/winsqlshell.cmd

```diff
@@ -0,0 +1,280 @@
+    REM error, print help and exit
+    CALL :help
+    EXIT /B 1
+)
+IF NOT EXIST "%SPLICE_LIB_DIR%\db-tools*.jar" (
+    ECHO Error: the db-tools required to connect cannot be found or created.
+    REM error, print help and exit
+    CALL :help
+    EXIT /B 1
+)
+
+SET CLASSPATH="%SPLICE_LIB_DIR%/*;";
+
+IF NOT [%HOSTARG%] == [] (
+   SET HOST=%HOSTARG%
+)
+IF NOT [%PORTARG%] == [] (
+   SET PORT=%PORTARG%
+)
+IF NOT [%USERARG%] == [] (
+   SET USER=%USERARG%
+)
+
+REM prompt securely for user password
+IF %PROMPT% EQU 1 (
+    REM use PowerShell to read password securely.
+    REM source: https://stackoverflow.com/a/20343074/337194
+    powershell -Command "$pword = read-host 'Enter password ' -AsSecureString ; $BSTR=[System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($pword) ; [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)" > .tmp.txt
+    SET /p PASS=<.tmp.txt & del .tmp.txt
+) ELSE IF NOT [%PASSARG%] == [] (
+   SET PASS="%PASSARG%"
+)
+
+REM Setup IJ_SYS_ARGS based on input options
+SET GEN_SYS_ARGS=-Djava.awt.headless^=true
+SET IJ_SYS_ARGS=-Djdbc.drivers^=com.splicemachine.db.jdbc.ClientDriver
+
+REM add width via ij.maximumDisplayWidth
+IF %WIDTH% NEQ 128 (
+   SET IJ_SYS_ARGS=%IJ_SYS_ARGS% -Dij.maximumDisplayWidth=%WIDTH%
+)
+
+IF NOT [%OUTPUT%] == [] (
+   REM figure out if OUTPUT directory exists
+   CALL :dir_from_path OUTPATH !OUTPUT!
+   IF NOT EXIST "%OUTPATH%" (
+     ECHO Error: you specified a non-existant directory for output %OUTPUT%
+     EXIT /B 1
+   )
+   SET IJ_SYS_ARGS=%IJ_SYS_ARGS% -Dij.outfile^=%OUTPUT%
+)
+
+IF NOT [%URL%] == [] (
+    SET IJ_SYS_ARGS=%IJ_SYS_ARGS% -Dij.connection.splice^=%URL%
+) ELSE (
+    REM Add optional URL parameters
+    SET SSL=""
+    SET KERBEROS=""
+    IF %SECURE% EQU 1 (
+        SET SSL=";ssl=basic"
+    )
+    IF NOT [%PRINCIPAL%] == [] (
+        SET KERBEROS=;principal^=%PRINCIPAL%
+        IF NOT [%KEYTAB%] == [] (
+            SET KERBEROS=%KERBEROS%;keytab^=%KEYTAB%
+        )
+    )
+    SET IJ_SYS_ARGS=%IJ_SYS_ARGS% -Dij.connection.splice^=jdbc:splice://%HOST%:%PORT%/splicedb%SSL%%KERBEROS%
+
+    SET JAVA_TOOL_OPTIONS=-Dij.user^=%USER% -Dij.password^=%PASS%
+    IF %QUOTE% EQU 1 (
+        SET JAVA_TOOL_OPTIONS=-Dij.user^='"%USER%"' -Dij.password^=%PASS%
+    )
+)
+
+IF NOT [%CLIENT_SSL_KEYSTORE%] == [] (
+    SET SSL_ARGS=-Djavax.net.ssl.keyStore^=%CLIENT_SSL_KEYSTORE% ^
+    -Djavax.net.ssl.keyStorePassword^=%CLIENT_SSL_KEYSTOREPASSWD% ^
+    -Djavax.net.ssl.trustStore^=%CLIENT_SSL_TRUSTSTORE% ^
+    -Djavax.net.ssl.trustStore.ssl.trustStorePassword^=%CLIENT_SSL_TRUSTSTOREPASSWD%
+)
+
+call :message Running Splice Machine SQL shell
+call :message For help%: "splice> help;"
+REM todo: catch stderr messages and filter out Picked up JAVA_TOOL_OPTIONS   2>nul
+java -classpath %CLASSPATH% %GEN_SYS_ARGS% %SSL_ARGS% %IJ_SYS_ARGS% com.splicemachine.db.tools.ij %SCRIPT% 2>nul
+
+CALL :eof
+

```

Comment:
this is returning and then printing ECHO %* again, resulting e.g. in
```
splice> exit;
-h srv131 -u bla -s password
```
at the end.

Reply:
Right, thanks, fixed it.